### DISCUSSION TITLE
pre-fill the the generated doc with some value

### DISCUSSION THREAD

**Discussion by sumskyi** [rswag/rswag#634] [General]
![image](https://user-images.githubusercontent.com/39335/235886853-3bde7f3a-15e4-426b-91b9-d837cb1de2e8.png)

My goal is using authorization token to use the `Execute` functionality in the generated API docs

the way how I am doing this (not documented as I see) 

`parameter name: :Authorization, in: :header, type: :string, required: true, value: 'Bearer: 23456'`

1. How could this be done via static values of a token? may be, fixtures, and hard-code token to the `value` ?
2. How to pass that authorization header into the real `curl` command?

**Comment by antonioJASR** [comment]
Hi, I think Swagger already has the implementation of bearer auth. So your use case can be resolved  following these steps:

1. define the authorization scheme under the components field (inside your `swagger_helper.rb` file):

```ruby
config.swagger_docs = {
  'v1/swagger.yaml' => {
    # ...
    components: {
      securitySchemes: {
        bearerAuth: { type: 'http', scheme: 'bearer'   } },
      # ...
    }
  }
}
```

2. Apply the authorization using the `security ` field in your Spec file:

```ruby
path '/api/v1/users/me' do
  get 'Who am I?' do
    tags 'Users'
    consumes 'application/json'
    security [bearerAuth: []] # <-- HERE

    response '200', 'Get the User' do
      # ...
      let(:access_token) { "GET_THE_ACCESS_TOKEN_SOMEWHERE" }
      let(:Authorization) { "Bearer #{access_token}" }
      run_test!
    end
  end
end

```


That's all. run `rake rswag:specs:swaggerize` and you will see in the UI the `AUTHORIZE BUTTON`

![Screenshot 2023-09-23 at 19 10 19](https://github.com/rswag/rswag/assets/1143701/035abd6d-6ebb-46f7-bcb4-49ae90e7fa01)

Clicking on that button this dialog will appear:

![Screenshot 2023-09-23 at 19 11 10](https://github.com/rswag/rswag/assets/1143701/a62a878d-b181-4535-9d0d-1b5337a76f40)

Hope it helps to anyone in the future :)

**Reply by sumskyi** [reply]
In my case it works this way:

```ruby
  let!(:jwt_token) { create(:jwt_token, user:) }
  let(:Authorization) { "Bearer #{jwt_token.token}" }
```

```ruby
  path '/api/v1/private/healthcheck' do
    get 'Makes a healthcheck' do
      tags 'Healthcheck'
      security [Bearer: {}] # it is a Hash declaration, not an Array

      produces 'application/json'
      consumes 'application/json'

      response '200', 'it is alive' do
        example 'application/json',
                :result,
                {
                  data: {
                    id: 'cad51544-f31b-4882-af50-2d29994ecdfb',
                    type: 'healthcheck',
                    attributes: {
                      id: 'cad51544-f31b-4882-af50-2d29994ecdfb',
                      time: '2023-05-04 12:09:24'
                    }
                  }
                }

        run_test! do
          expect(result_attributes[:id]).to be_present
          expect(result_attributes[:time]).to be_present
        end
      end
    end
  end
```
